hazelcast/src/main/java/com/hazelcast/map/impl/operation/steps/engine/CompositeStep.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation.steps.engine;
+
+import com.hazelcast.map.impl.operation.steps.IMapOpStep;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * The composite step for IMap operations. Composes multiple Steps into one unit.
+ * For instance, it is used to combine record store and local indexes compactors as one unit.
+ * <p>
+ * The primary step defines whether the composite step is offloadable, its executor name
+ * and next step.
+ */
+public class CompositeStep implements IMapOpStep {
```

Comment:
seems this class is no longer used.

Reply:
Removed https://github.com/hazelcast/hazelcast/pull/25208/commits/dcfc7309f9e91a71f62cfb4df4bd529b65de2340